cmd/di.go

```diff
@@ -0,0 +1,179 @@
+type Dependencies struct {
+	NodeOptions node.Options
+	Node        *node.Node
+
+	NetworkDefinition metadata.NetworkDefinition
+	MysteriumClient   server.Client
+	EtherClient       *ethclient.Client
+
+	Keystore         *keystore.KeyStore
+	IdentityManager  identity.Manager
+	SignerFactory    identity.SignerFactory
+	IdentityRegistry identity_registry.IdentityRegistry
+
+	IPResolver       ip.Resolver
+	LocationResolver location.Resolver
+
+	ServiceManager *service.Manager
+}
+
+// Bootstrap initiates all container dependencies
+func (di *Dependencies) Bootstrap(nodeOptions node.Options) error {
+	if err := nodeOptions.Directories.Check(); err != nil {
+		return err
+	}
+
+	if err := nodeOptions.Openvpn.Check(); err != nil {
+		return err
+	}
+
+	if err := di.bootstrapNetwork(nodeOptions.OptionsNetwork); err != nil {
+		return err
+	}
+
+	di.bootstrapIdentity(nodeOptions.Directories)
+	di.bootstrapLocation(nodeOptions.Location, nodeOptions.Directories.Config)
+	di.bootstrapNode(nodeOptions)
+
+	return nil
+}
+
+func (di *Dependencies) bootstrapNode(nodeOptions node.Options) {
+	di.NodeOptions = nodeOptions
+	di.Node = node.NewNode(
+		nodeOptions,
+		di.Keystore,
+		di.IdentityManager,
+		di.SignerFactory,
+		di.IdentityRegistry,
+		di.MysteriumClient,
+		di.IPResolver,
+		di.LocationResolver,
+	)
+}
+
+// BootstrapServiceManager initiates ServiceManager dependency
+func (di *Dependencies) BootstrapServiceManager(nodeOptions node.Options, serviceOptions service.Options) {
+	identityHandler := identity_selector.NewHandler(
+		di.IdentityManager,
+		di.MysteriumClient,
+		identity.NewIdentityCache(nodeOptions.Directories.Keystore, "remember.json"),
+		di.SignerFactory,
+	)
+	identityLoader := identity_selector.NewLoader(identityHandler, serviceOptions.Identity, serviceOptions.Passphrase)
+
+	di.ServiceManager = service.NewManager(
+		nodeOptions,
+		serviceOptions,
+		di.NetworkDefinition,
+		identityLoader,
+		di.SignerFactory,
+		di.IdentityRegistry,
+		di.MysteriumClient,
+		di.IPResolver,
+		di.LocationResolver,
+	)
+}
+
+// function decides on network definition combined from testnet/localnet flags and possible overrides
+func (di *Dependencies) bootstrapNetwork(options node.OptionsNetwork) (err error) {
+	network := metadata.DefaultNetwork
+
+	switch {
+	case options.Testnet:
+		network = metadata.TestnetDefinition
+	case options.Localnet:
+		network = metadata.LocalnetDefinition
+	}
+
+	//override defined values one by one from options
+	if options.DiscoveryAPIAddress != metadata.DefaultNetwork.DiscoveryAPIAddress {
+		network.DiscoveryAPIAddress = options.DiscoveryAPIAddress
+	}
+
+	if options.BrokerAddress != metadata.DefaultNetwork.BrokerAddress {
+		network.BrokerAddress = options.BrokerAddress
+	}
+
+	normalizedAddress := common.HexToAddress(options.EtherPaymentsAddress)
+	if normalizedAddress.String() != metadata.DefaultNetwork.PaymentsContractAddress.String() {
```

Comment:
Do we really need to convert both types to String? Struct should be comparable here

Reply:
Remade